java/runtime/src/main/java/io/ray/runtime/util/ConcurrencyGroupUtils.java

```diff
@@ -0,0 +1,90 @@
+package io.ray.runtime.util;
+
+import com.google.common.base.Preconditions;
+import io.ray.api.Ray;
+import io.ray.api.concurrencygroup.ConcurrencyGroup;
+import io.ray.api.concurrencygroup.annotations.DefConcurrencyGroup;
+import io.ray.api.concurrencygroup.annotations.DefConcurrencyGroups;
+import io.ray.api.concurrencygroup.annotations.UseConcurrencyGroup;
+import io.ray.api.function.RayFuncR;
+import io.ray.runtime.ConcurrencyGroupImpl;
+import io.ray.runtime.RayRuntimeInternal;
+import io.ray.runtime.functionmanager.JavaFunctionDescriptor;
+import java.lang.invoke.SerializedLambda;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/// TODO: cache this.
+public final class ConcurrencyGroupUtils {
+
+  public static List<ConcurrencyGroup> extractConcurrencyGroupsByAnnotations(
+      RayFuncR<?> actorConstructorLambda) {
+    SerializedLambda serializedLambda = LambdaUtils.getSerializedLambda(actorConstructorLambda);
+    Class<?> actorClz = getReturnTypeFromSignature(serializedLambda.getInstantiatedMethodType());
+
+    ArrayList<ConcurrencyGroup> ret = new ArrayList<ConcurrencyGroup>();
+    DefConcurrencyGroups concurrencyGroupsDefinitionAnnotation =
+        actorClz.getAnnotation(DefConcurrencyGroups.class);
+    if (concurrencyGroupsDefinitionAnnotation == null) {
+      /// This actor is not defined with concurrency groups definition.
+      return ret;
+    }
+
+    Map<String, ConcurrencyGroupImpl> concurrencyGroupsMap = new HashMap<>();
+    DefConcurrencyGroup[] defAnnotations = concurrencyGroupsDefinitionAnnotation.value();
+    if (defAnnotations.length == 0) {
+      throw new IllegalArgumentException("TODO");
```

Comment:
Will you address this TODO?

Reply:
It doesn't seem to happen this case. Let's use `checkState` instead.